design-decisions.md

```diff
@@ -0,0 +1,40 @@
+# `focus` attribute
+
+*You can contribute to the dicussion on this thread: https://github.com/aurelia/ux/issues/170*
+
+### API
+
+Inputable components should implement a `focus()` and `blur()` public API. Calling these methods must
+
+* display the component in its *focused* state
+* if text can be entered, it must focus the input tag, so that soft keyboard would open in mobile devices
+
+Also, if the inner HTML component receives focus (for exemple by the user hitting <TAB> key), the component must become in focused state.
+
+Since Aurelia has a built-in `focus` custom attribute, we expect Aurelia UX components to work seemlessly with this attribute. This constraints prevents us to use a `@bindable` for the focus attribute because the way Aurelia handles custom attributes conflicts with the bindable. The chosen solution is to add the `focus` and `blur` methods to the component's HTML element (through the custom prototype).
+
+### CSS
+
+When a component is in `focused` state, it must receive the CSS class `<comp-name>--focused`.
```

Comment:
pinging @EisenbergEffect @serifine for this change. At the moment, we are using `:focus` selector

Reply:
Would also like to have their input on this one. It seems that at the moment there is a mix of both `:focus` and `<comp-name>--focused`. I was thinking that we are trying to use more BEM conventions, hence this suggestion.